include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm_banded.hpp

```diff
@@ -85,18 +85,24 @@ class pairwise_alignment_algorithm_banded :
 
         for (auto && [sequence_pair, idx] : indexed_sequence_pairs)
         {
-            size_t const sequence1_size = std::ranges::distance(get<0>(sequence_pair));
+            size_t sequence1_size = std::ranges::distance(get<0>(sequence_pair));
             size_t const sequence2_size = std::ranges::distance(get<1>(sequence_pair));
 
+            auto && [alignment_matrix, index_matrix] = this->acquire_matrices(sequence1_size,
+                                                                              sequence2_size,
+                                                                              this->lowest_viable_score());
+
             // Initialise the cell updater with the dimensions of the regular matrix.
             this->compare_and_set_optimum.set_target_indices(row_index_type{sequence2_size},
                                                              column_index_type{sequence1_size});
 
-            auto && [alignment_matrix, index_matrix] = this->acquire_matrices(sequence1_size,
-                                                                              sequence2_size,
-                                                                              this->lowest_viable_score());
+            // Shrink the first sequence if the band ends before its actual end.
+            sequence1_size = std::min(sequence1_size, this->upper_diagonal + sequence2_size);
```

Comment:
Is it fine to first `set_target_indices` (line 96) of the score updater and **afterwards** changing the size of `sequence1`?

Reply:
yes, because in the acquire matrix algorithm we have to check if the band is valid, which we can only do on the original size. That means we might allocate a little to much memory, but that is usually not a problem since for the score we do not allocate the whole matrix and if we have the trace then I expect, that the difference is not by orders of magnitude.